pkg/core/xds/resource.go

```diff
@@ -146,10 +147,15 @@ func (s *ResourceSet) List() ResourceList {
 		return nil
 	}
 	list := ResourceList{}
-	list = append(list, s.ListOf(envoy_resource.EndpointType)...)
-	list = append(list, s.ListOf(envoy_resource.ClusterType)...)
-	list = append(list, s.ListOf(envoy_resource.RouteType)...)
-	list = append(list, s.ListOf(envoy_resource.ListenerType)...)
-	list = append(list, s.ListOf(envoy_resource.SecretType)...)
+	list = append(list, s.ListOf(envoy_resource_v2.EndpointType)...)
+	list = append(list, s.ListOf(envoy_resource_v3.EndpointType)...)
+	list = append(list, s.ListOf(envoy_resource_v2.ClusterType)...)
+	list = append(list, s.ListOf(envoy_resource_v3.ClusterType)...)
+	list = append(list, s.ListOf(envoy_resource_v2.RouteType)...)
+	list = append(list, s.ListOf(envoy_resource_v3.RouteType)...)
+	list = append(list, s.ListOf(envoy_resource_v2.ListenerType)...)
+	list = append(list, s.ListOf(envoy_resource_v3.ListenerType)...)
+	list = append(list, s.ListOf(envoy_resource_v2.SecretType)...)
+	list = append(list, s.ListOf(envoy_resource_v3.SecretType)...)
```

Comment:
nit: Hmm shall we split thes to v2/v3 specifics instead of mixing these side by side?